src/classes/LeadRepository.cls

```diff
@@ -24,12 +24,23 @@ public without sharing class LeadRepository extends SObjectRepository {
     public List<Lead> getList(List<Id> leadIdList) {
         String query = this
             .addConditionIdIn(leadIdList)
-            .setAsUpdate(true)
+            .setAsForUpdate()
             .getQuery();
 
         return (List<Lead>)Database.query(query);
     }
 
+    public List<Lead> searchInAllFields(String searchTerm) {
+        String query = this
+            .addConditionIsConverted(false)
+            .orderBy(Schema.Lead.CreatedDate, SObjectRepository.SortOrder.DESCENDING)
+            .limitCount(10)
+            .setAsForUpdate()
+            .getSearchQuery(searchTerm, SObjectRepository.SearchGroup.ALL_FIELDS);
+
+        return (List<Lead>)Search.query(query)[0];
```

Comment:
Dynamic SOSL should not be used as it could expose the application to SOSL injection security vulnerabilities.